plugins/Ecommerce/Columns/ProductCategory.php

```diff
@@ -10,10 +10,54 @@
 
 use Piwik\Columns\Dimension;
 use Piwik\Piwik;
+use Piwik\Plugin\Segment;
 
 class ProductCategory extends Dimension
 {
+    const PRODUCT_CATEGORY_COUNT = 5;
+
     protected $type = self::TYPE_TEXT;
     protected $category = 'Goals_Ecommerce';
     protected $nameSingular = 'Goals_ProductCategory';
+
+    protected function configureSegments()
+    {
+        $individualProductCategorySegments = $this->getProductCategorySegments(self::PRODUCT_CATEGORY_COUNT);
+
+        // add individual productCategoryN segments for use as a union (these segments are not available through the UI/API)
```

Comment:
But currently they are visible in UI :thinking:

Reply:
The comment's no longer accurate, I tried using `setPrivate(false)`, but then the segments wouldn't be available for the union of segments. I'm not sure of a way around this right now...

Reply:
Fixed thanks to suggestion from @tsteur